### PR TITLE
style: Button support gradient background

### LINKED ISSUE #40311: In v5, cannot define primary Buttons background (limited to background-color)
### What problem does this feature solve?

In antd v5, I would like to be able to use a gradient for the background of my primary Button.
For example: `transparent linear-gradient(180deg, #2CCEFF 0%, #0A8EE0 100%) 0% 0% no-repeat padding-box`

I'm able to change background color using a custom theme: changing the `colorPrimary` token define on `components.Button` is working. The problem is this token is affected to the `background-color` property in CSS, thus using a gradient is not working here. (see [source code](https://github.com/ant-design/ant-design/blob/master/components/button/style/index.ts#L241))

### What does the proposed API look like?

It would be nice to have access to the `background` property, instead of the `background-color` or via another token to be able to have more flexibility on backgrounds.

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #40311

ref https://github.com/ant-design/ant-design/pull/38644
ref https://github.com/ant-design/ant-design/pull/46188

### 💡 Background and solution

图片

图片

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Button could be customized to gradient style.       |
| 🇨🇳 Chinese |  Button 按钮支持自定义为渐变色风格。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/button/demo/linear-gradient.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Button: {
+          colorPrimary: 'linear-gradient(90deg, #FF4E50, #F9D423) !important',
+          primaryShadow: 'none',
+          defaultBg: 'linear-gradient(90deg, #aea4e3, #d3ffe8) !important',
+          defaultShadow: 'none',
```
[the remark this answers is not in the document]
Comment (afc163) [reply components/button/demo/linear-gradient.tsx:12]: 另外有 defaultBg 没有 primaryBg，感觉比较奇怪。

**Review by vagusX — APPROVED** [review]

**Comment by crazyair** [comment]
哈哈哈
image

File: components/button/demo/linear-gradient.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { Button, ConfigProvider, Space } from 'antd';
+
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Button: {
+          colorPrimary: 'linear-gradient(90deg, #FF4E50, #F9D423) !important',
+          primaryShadow: 'none',
+          defaultBg: 'linear-gradient(90deg, #aea4e3, #d3ffe8) !important',
```
[the remark this answers is not in the document]
Comment (yoyo837) [reply components/button/demo/linear-gradient.tsx:11]: 同反感

File: components/button/demo/linear-gradient.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (MadCcc) [reply components/button/demo/linear-gradient.tsx:12]: 这两点刚好矛盾了。。
没有 primaryBg 是因为可以直接改 colorPrimary，从语义上讲确实可以加

File: components/button/demo/linear-gradient.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (afc163) [reply components/button/demo/linear-gradient.tsx:11]: 不是覆盖不了，如果不加 important 就需要对 hover、active、focus 等多个背景色 token 做定制，非常麻烦。

[the remark this answers is not in the document]
Comment (crazyair) [reply components/button/demo/linear-gradient.tsx:11]: image

hover 本来就是不同色。那渐变场景下，hover 需要有渐变色的 hover 效果吗？

[the remark this answers is not in the document]
Comment (afc163) [reply components/button/demo/linear-gradient.tsx:11]: 这个得 case by case 看，貌似没有统一的处理方案。

**Pull request merged**